apps/sms/views/shared/js/settings.js

```diff
@@ -72,6 +79,15 @@ var Settings = {
     }
   },
 
+  // return resolve promise with sendReadReport status
+  initSendReadReport: function initSendReadReport(value) {
+    this._sendReadReportDefer.resolve(value);
+    delete this._sendReadReportDefer;
+
+    navigator.mozSettings.addObserver('messages.mms.sendReadReport.enabled',
+      (e) => this.sendReadReport = Promise.resolve(e.settingValue));
```

Comment:
Ok I see it's to prevent the request of sendReadReport start earlier than settings init ready.